src/instructlab/configuration.py

```diff
@@ -585,13 +608,154 @@ def ensure_storage_directories_exist():
+        },
+        ckpt_output_dir=os.path.join(DEFAULTS._data_dir, "checkpoints"),
+        data_output_dir=os.path.join(DEFAULTS._data_dir, "internal"),
+        data_path=os.path.join(DEFAULTS._data_dir, "datasets"),
+        nproc_per_node=4,
+        effective_batch_size=96,
+        lora_quantize_dtype=None,
+        lora_rank=0,
+        max_batch_len=60000,
+        max_seq_len=4096,
+        save_samples=1000,
+        deepspeed_cpu_offload_optimizer=True,
+        is_padding_free=True,
+        model_path=os.path.join(
+            DEFAULTS._cache_home, "models/instructlab/granite-7b-lab"
+        ),
+    )
+
+    EIGHT_L_FORTY_GPU = _train(
+        additional_args={
+            "warmup_steps": 10,
+            "learning_rate": 2e-5,
+            "lora_dropout": 0.1,
+            "lora_alpha": 32,
+        },
+        ckpt_output_dir=os.path.join(DEFAULTS._data_dir, "checkpoints"),
+        data_output_dir=os.path.join(DEFAULTS._data_dir, "internal"),
+        data_path=os.path.join(DEFAULTS._data_dir, "datasets"),
+        nproc_per_node=4,
+        effective_batch_size=96,
+        lora_quantize_dtype=None,
+        lora_rank=0,
+        max_batch_len=30000,
+        max_seq_len=4096,
+        save_samples=1000,
+        deepspeed_cpu_offload_optimizer=False,
+        is_padding_free=True,
+        model_path=os.path.join(
+            DEFAULTS._cache_home, "models/instructlab/granite-7b-lab"
+        ),
+    )
+
+    FOUR_L_FORTY_GPU = _train(
+        additional_args={
+            "warmup_steps": 10,
+            "learning_rate": 2e-5,
+            "lora_dropout": 0.1,
+            "lora_alpha": 32,
+            "deepspeed_cpu_offload_optimizer_ratio": 1,
+            "deepspeed_cpu_offload_optimizer_pin_memory": False,
+        },
+        ckpt_output_dir=os.path.join(DEFAULTS._data_dir, "checkpoints"),
+        data_output_dir=os.path.join(DEFAULTS._data_dir, "internal"),
+        data_path=os.path.join(DEFAULTS._data_dir, "datasets"),
+        nproc_per_node=4,
+        effective_batch_size=96,
+        lora_quantize_dtype=None,
+        lora_rank=0,
+        max_batch_len=30000,
+        max_seq_len=4096,
+        save_samples=1000,
+        deepspeed_cpu_offload_optimizer=True,
+        is_padding_free=True,
+        model_path=os.path.join(
+            DEFAULTS._cache_home, "models/instructlab/granite-7b-lab"
+        ),
+    )
+
+    EIGHT_L_FOUR_GPU = _train(
+        additional_args={
+            "warmup_steps": 10,
+            "learning_rate": 2e-5,
+            "lora_dropout": 0.1,
+            "lora_alpha": 32,
+            "deepspeed_cpu_offload_optimizer_ratio": 1,
+            "deepspeed_cpu_offload_optimizer_pin_memory": False,
+        },
+        ckpt_output_dir=os.path.join(DEFAULTS._data_dir, "checkpoints"),
+        data_output_dir=os.path.join(DEFAULTS._data_dir, "internal"),
+        data_path=os.path.join(DEFAULTS._data_dir, "datasets"),
+        nproc_per_node=4,
```

Comment:
Shouldn't this be 8 if we have 8 L40 GPUs?

Reply:
fixed